TESTS/mbed_hal_fpga_ci_test_shield/gpio/main.cpp

```diff
@@ -80,13 +85,23 @@ void test_basic_input_output(PinName pin)
     TEST_ASSERT_EQUAL_INT(0, gpio_read(&gpio));
     tester.gpio_write(MbedTester::LogicalPinGPIO0, 0, false);
     wait_us(HI_Z_READ_DELAY_US);
+    // Some targets don't support pull-up/pull-down mode.
+#if !defined(TARGET_NANO100) &&         \
+    !defined(TARGET_NUC472) &&          \
+    !defined(TARGET_M451)
     TEST_ASSERT_EQUAL_INT(1, gpio_read(&gpio)); // hi-Z, pulled up
+#endif
 
     // Test input, pull-down mode.
     gpio_mode(&gpio, PullDown);
     tester.gpio_write(MbedTester::LogicalPinGPIO0, 0, false);
     wait_us(HI_Z_READ_DELAY_US);
+    // Some targets don't support pull-up/pull-down mode.
```

Comment:
Same as above -- the whole `// Test input, pull-down mode.` should be excluded if pull-down is not supported.

Reply:
@fkjagodzinski Including above, fixed as commented

Reply:
Thanks for a quick update. Don't you think the `#if` statement is a bit too wide now? I thought that only the pull-up and pull-down modes are unsupported; pull-none was OK, wasn't it? I'd rather see [thses lines](https://github.com/ARMmbed/mbed-os/blob/398515a3678f78847118242a536de3ba0b058035/TESTS/mbed_hal_fpga_ci_test_shield/gpio/main.cpp#L70-L98) enclosed in `#if !defined(TARGET_NANO100) ...`. I do not have these targets available for local tests, so will let you decide. :wink:

Reply:
@fkjagodzinski Chip may have fixed (non-configurable) pull mode on these targets. To be clear, pull mode is not supported on these targets.

Reply:
In that case, we can leave the `#if` statements as they are.